vision_stack/task_buoy/src/scanningServer.py

```diff
@@ -0,0 +1,191 @@
+#! /usr/bin/env python
+
+import roslib
+PKG = 'task_buoy'
+roslib.load_manifest(PKG)
+
+import sys
+import numpy as np
+import time
+from math import *
+
+import actionlib
+import std_msgs.msg
+import rospy
+from resources import topicHeader
+# tools to get verbose
+from resources import tools
+
+from kraken_msgs.msg import thrusterData6Thruster
+from kraken_msgs.msg import thrusterData4Thruster
+from kraken_msgs.msg import absoluteRPY
+
+from kraken_msgs.msg import scanAction as actionMessagescanAction
+from kraken_msgs.msg import scanFeedback
+from kraken_msgs.msg import scanResult
+from kraken_msgs.msg import scanGoal
+
+# Global variables
+base_yaw = 0.0
+FIRST_ITERATION = True
+Current_yaw = -1
+
+class scanningServer(object):
+    """ Creates messages that are used to publish feedback/result to client
+        Messages :
+          *  _feedback :
+                1. Current value of yaw
+                2. Desired_yaw
+                3. Error
+          *  _result  :
+                1. elapsed_time
+          *  YawFeedback :
+                - Same as _feedback
+          *  thruster4Data :
+                - thrusterData4Thruster
+          *  thruster6Data :
+                - thrusterData6Thruster
+        Publishers :
+          *  pub_thrusters4 : publish as 4 thruster model on thruster converter
+          *  pub_thrusters4 : publish as 6 thruster model on thruster converter
+          *  pub_yawFeedback : Publish on 'YawFeedback' topic for plotting feedback
+          *  sub_ABSrpy : Subscribe absolute rpy from absoluteRPY publisher
+
+    """
+
+    # Initializing the messages to be used
+    _feedback = scanFeedback()
+    _result = scanResult()
+    thruster4Data=thrusterData4Thruster()
+    thruster6Data=thrusterData6Thruster()
+
+    def __init__(self, name):
+
+        self._action_name = name
+        self._as = actionlib.SimpleActionServer(
+            self._action_name, actionMessagescanAction, execute_cb=self.execute_cb, auto_start=False)
+        self._as.start()
+
+        # Declare all Publisher and Subscribers here :
+        self.pub_thrusters4 = rospy.Publisher(topicHeader.CONTROL_PID_THRUSTER4, thrusterData4Thruster, queue_size = 2)
+        self.pub_thrusters6 = rospy.Publisher(topicHeader.CONTROL_PID_THRUSTER6, thrusterData6Thruster, queue_size = 2)
+        self.pub_yawFeedback = rospy.Publisher('YawFeedback',scanFeedback, queue_size = 2)
+        self.sub_ABSrpy = rospy.Subscriber(topicHeader.ABSOLUTE_RPY, absoluteRPY, self.imuCB)
+
+        rospy.loginfo('Server has started. Waiting for a goal.')
+
+
+    def imuCB(self, dataIn):
+        """
```

Comment:
Yes, @aniket11bh is still working on this. me and @kalyan-kumar created the template and gave it to him.